client/reader/discover/controller.js

```diff
@@ -70,11 +86,25 @@ const exported = {
 				/>
 			</>
 		);
-		/* eslint-enable wpcalypso/jsx-classname-namespace */
+		next();
+	},
+
+	discoverSsr( context, next ) {
```

Comment:
You might want to put the `discover` and `discoverSsr` functions into separate files. Otherwise the browser bundle will include dead code for SSR and node bundle will include unneccesary code that's active only in the browser.

Here there is probably not so much code like this. I only see the `track*` functions from `controller-helper`. But it's still a good idea to keep them separate.

Reply:
Thanks, good point! I've moved the functions to their respective index file.